Documentation/network/clustermesh/services.rst

```diff
@@ -110,3 +109,50 @@ Deploying a Simple Example Service
       kubectl exec -ti deployment/x-wing -- curl rebel-base
 
    You will see replies from pods in both clusters again.
+
+Global and Shared Services Reference
+####################################
+
+The table below summarizes the overall behavior considering two clusters (i.e.,
+local and remote), and different combinations of the ``io.cilium/global-service``
+and ``io.cilium/shared-service`` annotation values.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Local Service Global
+     - Local Service Shared
+     - Remote Service Global
+     - Remote Service Shared
+     - Local Service Endpoints
+     - Remote Service Endpoints
+   * - Absent/False
+     - Don't care
+     - Don't care
+     - Don't care
```

Comment:
A little bit on wording, I am nt sure if there is better way. Let's see what others think.

Reply:
How about using a flowchart instead of a table since Global and Shared have dependencies?

```mermaid
flowchart LR
  LocalGlobal{LocalGlobal?}-->|yes|LocalShared{Local Shared?}
  LocalGlobal-->|no|LocalOnly[Local Only]
  LocalShared-->|yes|RemoteGlobal{Remote Global?}
  RemoteGlobal-->|yes|RemoteShared{Remote Shared?}
  RemoteShared-->|yes|LocalRemote[Local + Remote]
  RemoteGlobal-->|no|LocalOnly
  RemoteShared-->|no|LocalOnly
  LocalShared-->|no|RemoteGlobal2{Remote Global?}
  RemoteGlobal2-->|yes|RemoteShared2{Remote Shared?}
  RemoteGlobal2-->|no|LocalOnly
  RemoteShared2-->|yes|RemoteOnly[Remote Only]
  RemoteShared2-->|no|LocalOnly
```

Reply:
That looks really nice. Let me see if I can get it work in the docs.

Reply:
I made a few attempts but I couldn't get even the getting-started example of `sphinxcontrib-mermaid` to work properly. I'm wondering whether there might be some incompatibility with something in the theme.

Reply:
Needs more cycle? Then, maybe better to make this diagram stuff out of scope and move forward to merge the bug fix first.

Reply:
I've made some progress in getting mermaid flowcharts display in the docs, but there are still issues, and the process also adds different dependencies which bloat the builder image (especially for non-html targets). Since this will likely require additional discussion on the best way to get it done, I'm going to drop the documentation part from this PR, and open a new one for that.

Reply:
Ref: #23408